airbyte-api/src/main/openapi/config.yaml

```diff
@@ -3374,10 +3372,6 @@ components:
         prefix:
           type: string
           description: Prefix that will be prepended to the name of each stream when it is written to the destination.
-        operationIds:
```

Comment:
I'm a little confused why we are using `operations` in this `webBackendConnectionUpdate` endpoint, but we are using `operationIds` in the `connectionUpdate` endpoint. Is there a reason we don't want these to be consistent?

I'm also a bit confused as to why we have two "connection update" endpoints at all. Is the non-webBackend one just used by API users or something? What is the value we get out of having two different endpoints, instead of just consolidating them into one?

---
EDIT: This comment lower down in the code helped me understand this a bit better:
```
   * As a convenience to the front-end, this endpoint also creates new operations present in the
   * request, and bundles those newly-created operationIds into the connection update.
```

But it still seems like it would also be easier for API users to not have to create/update operations elsewhere and instead just do it in the updateConnection endpoint like the webBackend one does?

Reply:
I think the main idea here is that the public-facing endpoints like `connectionUpdate` should remain as "pure" as possible, meaning we only modify one resource at a time. I think we *want* API users to first use the `operationCreate` endpoint, and then pass the returned IDs into the `connectionUpdate` endpoint. This way, users who are creating systems and pipelines powered by our API are using very clear, discrete building blocks.

The `web_backend` routes violate this idea for the sake of performance and convenience

Reply:
Yeap, the general approach we have on the apis are:
1) pure resource routes - for programmatic access.
2) client specific routes (there is a better industry buzzword for this) - essentially convenience routes for various frontends. Today we only have the `webbackend` prefix.

Reply:
I think one of the reasons for web backend specific endpoints is also, that we often add "additional behavior" into them, e.g. figuring out if we should reset streams after updating, and in general if it would make sense to merge (in GET APIs) more resources together, so that the FE can have them in one call instead of two, we'd often change them (thus also no backward compatibility on them).

I do think there might be a reason for this "non pure" GET APIs, so we can optimize loading performance by having APIs that return us exactly what the FE needs even if it comes from separate APIs. Regarding the PATCH/POST (update) APIs, I feel maybe we could actually try to get them merged back into the "regular APIs" and have potential "additional behavior" just available via additional flags, e.g. as we already have with `skipReset` for the connection update, so we could tell the regular POST API which checks we'll need over the pure "saving of the resource", which most likely should be the default behavior.

Reply:
Got it, the motivation around having these as two separate endpoints makes sense. Though, I think there would be value in consolidating these through the use of flags as Tim suggested, so that we don't have duplication of API definitions which will inevitably lead to them becoming out of sync (e.g. an engineer makes a change to the base endpoint but forgets to change the webBackend one, causing a bug().

Reply:
The counterpoint to Tim's proposed approach is the consolidated endpoints become complicated as they start serving too many user types. This isn't likely to be great for the Cloud Api user segment. Because of this, prefer to keep separate for now.